Meadowlark-js/packages/meadowlark-core/src/middleware/ParsePathMiddleware.ts

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+import { writeDebugStatusToLog, writeRequestToLog } from '../Logger';
+import { isDocumentIdValid } from '../model/DocumentId';
+import type { PathComponents } from '../model/PathComponents';
+import type { MiddlewareChain } from './MiddlewareChain';
+
+const moduleName = 'ParsePathMiddleware';
+
+function pathComponentsFrom(path: string): PathComponents | null {
+  // Matches all of the following sample expressions:
```

Comment:
moved the pathComponentsFrom function here, since this was the only code using it. Then I realized it meant I had to test it. 😆